libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -1893,6 +1893,86 @@ namespace meshkernelapi
         return lastExitCode;
     }
 
+    MKERNEL_API int mkernel_mesh2d_get_face_polygons(int meshKernelId, int numNodes, GeometryList& facePolygons)
+    {
+        lastExitCode = meshkernel::ExitCode::Success;
+        try
+        {
+            if (!meshKernelState.contains(meshKernelId))
+            {
+                throw meshkernel::MeshKernelError("The selected mesh kernel id does not exist.");
+            }
+            meshKernelState[meshKernelId].m_mesh2d->Administrate();
+            const auto numFaces = meshKernelState[meshKernelId].m_mesh2d->GetNumFaces();
+            meshkernel::UInt count = 0;
+            for (meshkernel::UInt f = 0u; f < numFaces; ++f)
+            {
+                const auto& faceNodes = meshKernelState[meshKernelId].m_mesh2d->m_facesNodes[f];
+                const auto faceNodesSize = static_cast<int>(faceNodes.size());
+                if (faceNodesSize != numNodes)
+                {
+                    continue;
+                }
+                if (count != 0)
+                {
+                    facePolygons.coordinates_x[count] = missing::doubleValue;
+                    facePolygons.coordinates_y[count] = missing::doubleValue;
+                    count++;
+                }
+
+                for (meshkernel::UInt n = 0u; n < faceNodes.size(); ++n)
+                {
+                    const auto currentNode = meshKernelState[meshKernelId].m_mesh2d->Node(faceNodes[n]);
+                    facePolygons.coordinates_x[count] = currentNode.x;
+                    facePolygons.coordinates_y[count] = currentNode.y;
+                    count++;
+                }
+                const auto currentNode = meshKernelState[meshKernelId].m_mesh2d->Node(faceNodes[0]);
+                facePolygons.coordinates_x[count] = currentNode.x;
+                facePolygons.coordinates_y[count] = currentNode.y;
+                count++;
+            }
+        }
+        catch (...)
+        {
+            lastExitCode = HandleException();
+        }
+        return lastExitCode;
+    }
+
+    MKERNEL_API int mkernel_mesh2d_get_face_polygons_dimension(int meshKernelId, int numNodes, int& geometryListDimension)
+    {
+        lastExitCode = meshkernel::ExitCode::Success;
+        try
+        {
+            if (!meshKernelState.contains(meshKernelId))
+            {
+                throw meshkernel::MeshKernelError("The selected mesh kernel id does not exist.");
+            }
+            meshKernelState[meshKernelId].m_mesh2d->Administrate();
+            const auto numFaces = meshKernelState[meshKernelId].m_mesh2d->GetNumFaces();
```

Comment:
`static_cast<meshkernel::UInt>(meshKernelState[meshKernelId].m_mesh2d->GetNumFaces())`

Reply:
No need to cast to UInt, GetNumFaces return type is already UInt